### DISCUSSION TITLE
WandB Logging DDP

### DISCUSSION THREAD

**Discussion by benjaminrwilson** [Lightning-AI/pytorch-lightning#7316] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
What is the proper way to prevent repeated WandB experiments when using DDP with multiple GPUs?

The WandB discussion discusses it here: https://docs.wandb.ai/guides/track/advanced/distributed-training (Method 1). Has anyone had any success preventing each GPU spawning an experiment?

Thanks in advance!

**Comment by awaelchli** [comment]
You can use our [wrapper for wandb](https://pytorch-lightning.readthedocs.io/en/latest/extensions/generated/pytorch_lightning.loggers.WandbLogger.html#pytorch_lightning.loggers.WandbLogger). 

It logs only on the first process, so only one experiment gets created.
```python
from pytorch_lightning.loggers import WandbLogger
logger = WandbLogger(project="the coconut", name="is not a nut")
trainer = Trainer(logger=logger)
```

**Reply by benjaminrwilson** [reply]
Thanks for the response, @awaelchli. I am using the built in wrapper for Wandb. For some more context, my jobs are being submitted to a SLURM cluster via the `submitit` plugin using the `--multirun` option.

For example, here is a 2-GPU job I submitted:

<img width="408" alt="image" src="https://user-images.githubusercontent.com/20147747/116832940-c55a6300-ab84-11eb-87fa-d65fc7f0b28b.png">

which causes each GPU to be logged. I'm using groups to at least keep the individual runs together. Maybe I have another setting set incorrectly?

**Reply by awaelchli** [reply]
It could be that you are using a lightning version where we had bugs surrounding the trainer not setting the rank information correctly when using SLURM. Make sure to test the latest Lightning release. 

Are both experiments logging values? I suspect that one of them is empty.

**Reply by benjaminrwilson** [reply]
I'm using a fork I pulled from the main branch yesterday. Yeah, only the rank zero experiment is logging. There are no functional issues per se, but the two downsides are:

1. Cluttered dashboard.
2. Only the rank zero experiment shows `finished` after completing the job. All of the other experiments (from other gpus) show `crashed`.
